### DISCUSSION TITLE
Overlaying hover pop ups and location of the "Run" window.

### DISCUSSION THREAD

**Discussion by oc-tk** [mrcjkb/rustaceanvim#125] [Q&A] [answered]
Hi!
At start I have to say that I am new to the the Neovim and configuring lua pluings. So sorry if I made a mess.
I am configuring my Neovim config (made on kickstarter.nvim, this is my [fork)](https://github.com/oc-tk/kickstart.nvim)) and I have installed rustaceanvim and I am fighting with run window position (I would like to move it to below) and sometimes it displays just function name (mainly main) without option to Run / Debug it. Sometimes I can call both of them.
<img width="737" alt="hover_bug" src="https://github.com/mrcjkb/rustaceanvim/assets/8511623/3cfdceb2-82ec-4ec6-b8bf-30c95427813d">

Edit: Not hover hint position, position of the run window

**Comment by mrcjkb** [comment]
Hey 👋 

About the position: I'm not sure if it's possible to change that. I'd look at [`:h vim.lsp.buf.hover`](https://neovim.io/doc/user/lsp.html#vim.lsp.buf.hover()), but it doesn't look like it has any customisation options.

Otherwise, I'd recommend asking [the Neovim community](https://neovim.io/community/) - this sounds like it's not specific to this plugin.

About the second issue:

rust-analyzer tries to detect if it can run or debug a function and if it can, it'll add an option to the hover response.
That's not always the case. Especially during startup, rust-analyzer might send a response without run/debug options if it's still analyzing your project.

**Reply by oc-tk** [reply]
I was doing this after rust-analyzer have runned checks etc. I have autocompletion and lsp working and it is not adding option in hover response.

**Reply by oc-tk** [reply]
And sorry, I made an mistake. In first problem I want to move output window/tab with build / run output to bottom/side.
<img width="1160" alt="image" src="https://github.com/mrcjkb/rustaceanvim/assets/8511623/ed2ad4f7-9e70-4b1d-956f-ac1f8a2f84e6">

**Reply by mrcjkb** [reply]
> I was doing this after rust-analyzer have runned checks etc. I have autocompletion and lsp working and it is not adding option in hover response.

In that case I guess rust-analyzer just can't find any runnables/debuggables for your function.

> In first problem I want to move output window/tab with build / run output to bottom/side.

I'm not sure I understand. That sounds like a [nvim-dap](https://github.com/mfussenegger/nvim-dap) issue?

**Reply by oc-tk** [reply]
> In that case I guess rust-analyzer just can't find any runnables/debuggables for your function.
Is there a way to force rust-analyzer to always override main function?

> I'm not sure I understand. That sounds like a [nvim-dap](https://github.com/mfussenegger/nvim-dap) issue?
Okay, it looks like you are right. I will look into it

**Reply by mrcjkb** [reply]
> Is there a way to force rust-analyzer to always override main function?

Not that I know of

**Reply by oc-tk** [reply, marked as answer]
Alright I have solder this.
So to have terminal you have to use different split option. For my case it was `vim.o.splitbelow = true`.
For second thing. LSPzero has problem with rust-analyzer/OG LSP and there is this overriding of hover. I have disabled LSP zero and it workss ifne.
